src/upload/node.ts

```diff
@@ -0,0 +1,262 @@
+ * @returns - The upload response.
+ * @throws - Will throw if the directory is invalid.
+ */
+export async function uploadDirectoryFromPathRequest(
+  this: SkynetClient,
+  path: string,
+  customOptions = {}
+): Promise<AxiosResponse> {
+  const opts = { ...defaultUploadOptions, ...this.customOptions, ...customOptions };
+
+  // Check if there is a directory at given path.
+  const stat = fs.statSync(path);
+  if (!stat.isDirectory()) {
+    throw new Error(`Given path is not a directory: ${path}`);
+  }
+
+  const formData = new FormData();
+  path = p.normalize(path);
+  let basepath = path;
+  // Ensure the basepath ends in a slash.
+  if (!basepath.endsWith("/")) {
+    basepath += "/";
+    // Normalize the slash on non-Unix filesystems.
+    basepath = p.normalize(basepath);
+  }
+
+  for (const file of walkDirectory(path)) {
+    // Remove the dir path from the start of the filename if it exists.
+    let filename = file;
+    if (file.startsWith(basepath)) {
+      filename = file.replace(basepath, "");
+    }
+    formData.append(opts.portalDirectoryFileFieldname, fs.createReadStream(file), { filepath: filename });
+  }
+
+  // TODO: Implement customDirname
+  // let filename = opts.customDirname || path;
+  let filename = path;
+  /* istanbul ignore next */
+  if (filename.startsWith("/")) {
+    filename = filename.slice(1);
+  }
+
+  return this.executeRequest({
+    ...opts,
+    method: "post",
+    // @ts-expect-error TS doesn't recognize this external FormData.
+    data: formData,
+    headers: formData.getHeaders(),
+    query: { filename },
+  });
+}
+
+/**
+ * Uploads a file to Skynet.
+ *
+ * @param this - SkynetClient
+ * @param fileContents - The file contents to upload.
+ * @param fileName - The desired name for the file.
+ * @param [customOptions] - Additional settings that can optionally be set.
+ * @param [customOptions.endpointPath="/skynet/skyfile"] - The relative URL path of the portal endpoint to contact.
+ * @returns - The returned skyfile information including skylink, merkleroot and bitfield.
+ * @throws - Will throw if the request is successful but the upload response does not contain a complete response.
+ */
+export async function uploadFileContent(
+  this: SkynetClient,
+  fileContents: string,
+  fileName: string,
+  customOptions?: CustomUploadOptions
+): Promise<UploadRequestResponse> {
+  const response = await this.uploadFileContentRequest(fileContents, fileName, customOptions);
+
+  /* istanbul ignore next */
+  if (
+    typeof response.data.skylink !== "string" ||
+    typeof response.data.merkleroot !== "string" ||
+    typeof response.data.bitfield !== "number"
+  ) {
+    throw new Error(
+      "Did not get a complete upload response despite a successful request. Please try again and report this issue to the devs if it persists."
+    );
+  }
+
+  const skylink = formatSkylink(response.data.skylink);
+  const merkleroot = response.data.merkleroot;
+  const bitfield = response.data.bitfield;
```

Comment:
optional tip: it's handy to use object destructuring assignment in such cases

```suggestion
  const { merkleroot, bitfield } = response.data;
```